src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -351,21 +351,26 @@ void QGIDatumLabel::hoverLeaveEvent(QGraphicsSceneHoverEvent* event)
 
 QRectF QGIDatumLabel::boundingRect() const
 {
-    return childrenBoundingRect();
+    QRectF totalRect;
+    for (QGraphicsItem* item : m_textItems->childItems()) {
+        if (auto* customText = dynamic_cast<QGCustomText*>(item)) {
+            QRectF itemRect = customText->alignmentRect();
+            QPointF pos = customText->pos();
+            itemRect.translate(pos.x(), pos.y());
+            totalRect = totalRect.isNull() ? itemRect : totalRect.united(itemRect);
+        }
+    }
+    int fontSize = m_dimText->font().pixelSize();
+    int paddingLeft = fontSize * 0.2;
+    int paddingTop = fontSize * 0.1;
+    int paddingRight = fontSize * 0.2;
+    int paddingBottom = fontSize * 0.1;
+    return totalRect.adjusted(-paddingLeft, -paddingTop, paddingRight, paddingBottom);
```

Comment:
I don't think redefining `childrenBoundingRect` calculations are the right way to go.

Instead, the `boundingRect` function of the child to return `alignmentRect`. And then all `childrenBoundingRect` will be natively correct. I find this important for consistency in our code.

However, didn't `boundingRect` use to return `alignmentRect`?

Please explain if my proposed solution is not possible.

Reply:
see PR #18771.  Making the boundingRect smaller will cause graphics issues.

Reply:
Are we sure, that we won't see the same issues by redefining `boundingRect` when we redefine `childrenBoundingRect`? Maybe naming this function `childrenAdjustmentRect`?

I don't really have the nescarry overview of all the text alignment code.

Reply:
The child graphics items are no longer clipped, I tried out a bunch of the fonts that are available on my system (macOS), most seemed to work pretty well, but there were a couple like Bangla MN, and Academy Engraved LET which are drawn towards the top of their bounding box and don't really render perfectly. As was mentioned in #11452 I'm not sure there's an easy way around this. I guess you _could_ play waka-mole inside of `QGCustomText::tightBoundingRect()` adjusting the tight bounding box based on the font name?
<img width="213" alt="Screenshot 2025-01-22 at 6 10 28 PM" src="https://github.com/user-attachments/assets/fa2a1532-d313-4b4e-96f8-cfd914b9fd49" />